### DISCUSSION TITLE
libfwupdplugin1 issue when upgrading packages

### DISCUSSION THREAD

**Discussion by colmexdev** [fwupd/fwupd#4496] [Q&A]
**Describe the question**
Hello there.

   I've recently updated my Ubuntu 20.04 packages and received this message from apt:

   ```
   The following package was automatically installed and is no longer required:
   libfwupdplugin1
   Use 'apt-get autoremove' to remove it.
   The following packages have been kept back:
   postgresql-server-dev-13
   ...
   ```

   What does it mean that libfwupdplugin1 keeps back the postgresql-server-dev-13 package from upgrading? What can I do to solve it?

   I'm newbie to Ubuntu, so I apologize if this is a fairly easy/non-related question.

   Thanks in advance for your help!

**fwupd version information**
runtime   org.freedesktop.fwupd         1.7.5
compile   org.freedesktop.gusb          0.3.4
runtime   com.dell.libsmbios            2.4
runtime   org.kernel                    5.4.0-107-generic
compile   org.freedesktop.fwupd         1.7.5
runtime   org.freedesktop.gusb          0.3.4

**Comment by hughsie** [comment]
@superm1 any ideas what happened? I'm a Debian newbie.

**Comment by superm1** [comment]
The two events have nothing to do with each other. 

> The following package was automatically installed and is no longer required:
libfwupdplugin1

This means that you no longer have reverse dependencies installed that need this package and it wasn't manually installed.  It probably happened from an upgrade of fwupd from fwupd 1.3.x to 1.7.x which happened in Ubuntu 20.04 recently.  You can run `sudo apt autoremove` and it will remove it and you should be fine.

> The following packages have been kept back:
postgresql-server-dev-13

This means something about how you can't upgrade this package due to other stuff on the system.  You can try
`sudo apt dist-upgrade` to see if it tries to let it upgrade.  OT for fwupd though, you might try stack overflow or similar.
